assets/app/view/game/abilities.rb

```diff
@@ -94,6 +95,20 @@ def actions_for(company)
         @game.round.actions_for(company)
       end
 
+      # Render a button for the purchase train action that purchases the
+      # currently available depot train
+      def render_purchase_train_button
```

Comment:
is this needed? why not just use a choice?

Reply:
This is a new company action that lets the owning corporation buy the top train of the depot for normal price at any time. Added here: https://github.com/kelsin/18xx/blob/g_1871/lib/engine/ability/README.md#purchase_train